core/spec/models/spree/address_spec.rb

```diff
@@ -23,6 +23,33 @@
 
       expect(address.name).to eq('Ryan Small')
     end
+
+    it 'updates firstname and lastname when updating name' do
+      address.name = 'Jane Von Doe'
+
+      expect(address.firstname).to eq('Jane')
+      expect(address.lastname).to eq('Von Doe')
+    end
+  end
```

Comment:
Can we add a test when both `firstname`, `lastname` and `name` are tried to be set in the same instance? I know it should not happen but I'd like to anticipate a default behavior for people messing up this thing in their own applications.

Reply:
Yes, absolutely 👍

Reply:
Having a default here is tricky. The ways I can think of to update the three attributes in a single method call are AR methods taking an hash in input like  `attributes=`, `update` and `update_columns`.
`attributes=` and `update` use model accessors and run trough the input hash in sequence so the final outcome will be different depending on the order of the elements in the hash. 
`update_columns` skips accessors and could write non coherent `name`, `firstname` and `lastname` values.
I think that overriding all AR setters to handle corner cases will be overkill and unmaintainable, another solution could be to use `after_save` hooks but I don't like it either. I'm more inclined to leaving it this way, WDYT?

Reply:
what about removing `firstname` and `lastname` from attributes if `name` is also in there? Is it somehow possible at a model level?

Reply:
We could decorate `ActiveModel::ForbiddenAttributesProtection#sanitize_for_mass_assignment` and reject them if `name` is present. `sanitize_for_mass_assignment` is called during every mass assignment, it should work everywhere and it's not too invasive.